CMakeLists.txt

```diff
@@ -1,187 +1,311 @@
+option(SINGULARITY_USE_SPINER "Use spiner EOS" OFF)
+cmake_dependent_option(
+  SINGULARITY_USE_SPINER_WITH_HDF5 "Use HDF5 component of spiner" ON
+  "SINGULARITY_USE_SPINER" OFF)
+
+option(SINGULARITY_USE_FORTRAN "Enable fortran bindings" ON)
+option(SINGULARITY_USE_KOKKOS "Use Kokkos for portability" OFF)
+option(SINGULARITY_USE_EOSPAC "Pull in eospac" OFF)
+
+cmake_dependent_option(SINGULARITY_USE_CUDA "Use CUDA backend of Kokkos" ON
+                       "SINGULARITY_USE_KOKKOS" OFF)
+cmake_dependent_option(
+  SINGULARITY_USE_KOKKOSKERNELS "Use KokkosKernels for LA routines" ON
+  "SINGULARITY_USE_KOKKOS" OFF)
+cmake_dependent_option(
+  SINGULARITY_BUILD_CLOSURE "Mixed Cell Closure" ON
+  "SINGULARITY_USE_KOKKOSKERNELS;SINGULARITY_USE_KOKKOS" OFF)
+
+# extra build options
+option(SINGULARITY_BUILD_PYTHON "Compile Python bindings" OFF)
+option(SINGULARITY_BUILD_EXAMPLES "Compile examples" OFF)
+cmake_dependent_option(
+  SINGULARITY_BUILD_SESAME2SPINER
+  "Compile sesame2spiner"
+  ON
+  "SINGULARITY_USE_SPINER;SINGULARITY_USE_SPINER_WITH_HDF5;SINGULARITY_USE_EOSPAC"
+  OFF)
+cmake_dependent_option(
+  SINGUALRITY_BUILD_STELLARCOLLAPSE2SPINER "Compiler stellarcollapse2spiner" ON
+  "SINGULARITY_USE_SPINER;SINGULARITY_USE_SPINER_WITH_HDF5" OFF)
+
+# testing options
+option(SINGULARITY_BUILD_TESTS "Compile tests" OFF)
+
+cmake_dependent_option(
+  SINGULARITY_TEST_SESAME "Test Sesame table readers" ON
+  "SINGULARITY_BUILD_TESTS;SINGULARITY_BUILD_SESAME2SPINER" OFF)
+cmake_dependent_option(
+  SINGULARITY_TEST_STELLAR_COLLAPSE "Test stellar collapse table readers" ON
+  "SINGULARITY_BUILD_TESTS;SINGULARITY_BUILD_STELLARCOLLAPSE2SPINER" OFF)
+cmake_dependent_option(SINGULARITY_TEST_PYTHON "Test the Python bindings" ON
+                       "SINGULARITY_BUILD_TESTS;SINGULARITY_BUILD_PYTHON" OFF)
+
+# modify flags options
+option(SINGULARITY_BETTER_DEBUG_FLAGS "Better debug flags for singularity" ON)
+option(SINGULARITY_HIDE_MORE_WARNINGS "hide more warnings" OFF)
+
+# toggle code options
+option(SINGULARITY_USE_SINGLE_LOGS
+       "Use single precision logs. Can harm accuracy." OFF)
+option(SINGULARITY_USE_TRUE_LOG_GRIDDING
+       "Use grids that conform to log spacing." OFF)
+
+# misc options
+option(SINGULARITY_FORCE_SUBMODULE_MODE "Submodule mode" OFF)
+# TODO This is an edge-case, but still used (e.g. github CI), but need to work
+# out a way to do SUBMODULE_MODE (use submodules for upstream deps) but also do
+# a proper install (FORCE_ disables the install/export stages). I may come back
+# to this, but there isn't a lot of clamoring for this right now so could return
+# to it in a later PR. something like
+# cmake_dependent_option(SINGULARITY_SUBMODULE_MODE_INSTALL ...)
+
+# TODO This should depend on using offloading (`SINGULARITY_USE_CUDA` &t) Also,
+# patching in the source tree is messy, this should be done in the build dir
+option(SINGULARITY_PATCH_MPARK_VARIANT
+       "Apply GPU patch to mpark-variant submodule" ON)
```

Comment:
Our patch is (and will remain) compatible with all necessary backends via preprocessor. It is able to be patched unconditionally.

Reply:
Should it just always be patched then, without an option at all?

Reply:
I think it would be good to do so such that we always maintain compatibility, but we can save this for later if needed (if we wait for later, we should make an issue).